### ISSUE TITLE
tap container fails if pods do not belong a deployment

### ISSUE THREAD

**Issue by FaKod** [runconduit/conduit#80]
conduit v0.1.1

```
~> kubectl -n conduit logs controller-2041965127-1kflh tap
serving scrapable metrics on :9998
time="2017-12-21T05:56:49Z" level=info msg="starting gRPC server on :8088"
E1221 05:56:49.374317       1 runtime.go:66] Observed a panic: "index out of range" (runtime error: index out of range)
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:72
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:65
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:509
/usr/local/go/src/runtime/panic.go:491
/usr/local/go/src/runtime/panic.go:28
/go/src/github.com/runconduit/conduit/controller/k8s/replicasets.go:77
/go/src/github.com/runconduit/conduit/controller/tap/server.go:336
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/thread_safe_store.go:232
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/thread_safe_store.go:129
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/store.go:217
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/reflector.go:332
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/reflector.go:256
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/reflector.go:198
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait/wait.go:96
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait/wait.go:97
/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait/wait.go:52
/usr/local/go/src/runtime/asm_amd64.s:2337
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 13 [running]:
github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:58 +0x111
panic(0x14002c0, 0x1ef4020)
	/usr/local/go/src/runtime/panic.go:491 +0x283
github.com/runconduit/conduit/controller/k8s.(*ReplicaSetStore).GetDeploymentForPod(0xc4203cd280, 0xc420618c10, 0x0, 0xc42050d870, 0x8ba2eb, 0x141c6a0)
	/go/src/github.com/runconduit/conduit/controller/k8s/replicasets.go:77 +0x3aa
github.com/runconduit/conduit/controller/tap.NewServer.func1(0x156d480, 0xc420618c10, 0xc42050d918, 0x1, 0xc42090e178, 0x0, 0x0)
	/go/src/github.com/runconduit/conduit/controller/tap/server.go:336 +0x53
github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache.(*threadSafeMap).updateIndices(0xc42024b5c0, 0x0, 0x0, 0x156d480, 0xc420618c10, 0xc4208e0750, 0x2e, 0x0, 0x0)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/thread_safe_store.go:232 +0x265
github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache.(*threadSafeMap).Replace(0xc42024b5c0, 0xc420182180, 0xc42007e0e0, 0x7)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/thread_safe_store.go:129 +0x171
github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache.(*cache).Replace(0xc4203cd300, 0xc42037e000, 0x8d, 0x8d, 0xc42007e0e0, 0x7, 0x1, 0x7fed99d9f010)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/store.go:217 +0x1d8
github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache.(*Reflector).syncWith(0xc4202e7680, 0xc420488000, 0x8d, 0x8d, 0xc42007e0e0, 0x7, 0x0, 0x0)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/reflector.go:332 +0x1c3
github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc4202e7680, 0xc42009c2a0, 0x0, 0x0)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/reflector.go:256 +0x722
github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil.func1()
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/reflector.go:198 +0x33
github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc4203cd360)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait/wait.go:96 +0x5e
github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc4203cd360, 0x3b9aca00, 0x0, 0x1, 0xc42009c2a0)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait/wait.go:97 +0xa1
github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc4203cd360, 0x3b9aca00, 0xc42009c2a0)
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/pkg/util/wait/wait.go:52 +0x4d
created by github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil
	/go/src/github.com/runconduit/conduit/vendor/k8s.io/client-go/tools/cache/reflector.go:197 +0x18d
```

**Comment by klingerf** [comment]
@FaKod Thanks for reporting, and for submitting the fix. I think this is a dupe of #33, so will close that one out in favor of this one.

**Comment by wmorgan** [comment]
This was fixed in #81.
